### ISSUE TITLE
The InferenceService "sklearn-iris" is invalid: metadata.managedFields.fieldsType: Invalid value: "": must be `FieldsV1`

### ISSUE THREAD

**Issue by kmr0877** [kubeflow/kfserving#922]
/kind bug

**What steps did you take and what happened:**
I am trying to test KFServing Installation, setting up for the first time from [Test KFServing Installation](https://github.com/kubeflow/kfserving#test-kfserving-installation) .
Not really sure if its a bug?

Commands shown below
```sh
kubectl create namespace kfserving-test
kubectl apply -f docs/samples/sklearn/sklearn.yaml -n kfserving-test
```

Error message:
```sh
The InferenceService "sklearn-iris" is invalid: metadata.managedFields.fieldsType: Invalid value: "": must be `FieldsV1`
```

**What did you expect to happen:**
```sh
$ inferenceservice.serving.kubeflow.org/sklearn-iris created
```

**Anything else you would like to add:**
- I have got istio installed with manifest from [Installing Istio without sidecar injection](https://knative.dev/docs/install/installing-istio/#installing-istio-without-sidecar-injection) and Knative components using [Operators](https://knative.dev/v0.14-docs/install/knative-with-operators/)
- Not sure if its an issue with any of the versions i am using?
```sh
$ kubectl get pods --namespace knative-serving
NAME                                READY   STATUS     RESTARTS   AGE
activator-6d9f95b7f8-k5gtd          2/2     Running    0          24m
autoscaler-597fd8d69d-b6htb         2/2     Running    0          24m
autoscaler-hpa-5878b57d96-fvrm7     2/2     Running    1          24m
controller-7479cc984d-m7dxh         2/2     Running    1          24m
istio-webhook-64b48869b8-nmckd      2/2     Running    1          24m
networking-istio-7bb58f6d55-2hlgw   1/1     Running    0          24m
storage-version-migration-bcq59     1/2     NotReady   2          24m
webhook-bf465f954-2nbq9             2/2     Running    1          24m
```

**Environment:**

- Istio Version:
```sh
$ istioctl version
client version: 1.6.4
control plane version: 1.6.4
data plane version: 1.6.4 (10 proxies)
```

- Knative Version:
```
v0.15.0
```
- KFServing Version:
```sh
$ kubectl get KnativeServing knative-serving -n knative-serving
NAME              VERSION   READY   REASON
knative-serving   0.15.0    True
```
- Kubeflow version:
NA

- Kfdef:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
NA
- Minikube version:
```sh
$ minikube  version
minikube version: v1.11.0
commit: 57e2f55f47effe9ce396cea42a1e0eb4f611ebbd
```

- Kubernetes version: (use `kubectl version`):
```sh
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.5", GitCommit:"e6503f8d8f769ace2f338794c914a96fc335df0f", GitTreeState:"clean", BuildDate:"2020-06-26T03:47:41Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:43:34Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
``
- OS (e.g. from `/etc/os-release`):
```sh
Edition          Windows 10 Enterprise
Version             1903
Installed on       24/01/2020
OS build            18362.778

**Comment by yuzisun** [comment]
@kmr0877 you are on kubernetes 1.18 and looks like this is the same issue as https://github.com/GoogleCloudPlatform/flink-on-k8s-operator/issues/210, if you downgrade to 1.17 that should work. We will need to upgrade k8s client dependency for KFServing controller.

**Comment by kmr0877** [comment]
@yuzisun Seems like on downgrade to 1.17 works fine for me. Thanks for your response.
Worth upgrading k8s client, probably better to keep the issue open until its upgraded else we might end up seeing duplication as you mentioned above.
Feel free to close the issue if you have a different opinion.

**Comment by yuzisun** [comment]
released in 0.4.1
